### ISSUE TITLE
Recent release of apg and apg-lib breaks all old versions 😭

### ISSUE THREAD

**Issue by LinusU** [ldthomas/apg-js2-exp#2]
Since the dependencies was specified as `*`, the new release of these libs breaks all the 1.x versions of this library. I would advise to publish a new 1.x release with the dependencies looked to a version that works, e.g. `"apg": "^2.2.2", "apg-lib": "^2.1.5"`.

LPT: if you just use `npm install --save _` and never modify the package.json manually, this will be provided automatically for you

**Comment by ldthomas** [comment]
Oops. I see the problem. Using "*" in 1.0.5 was an oversight. What I don't see is how to fix it other than to push up a new README file advising users of version 1.0.5, or earlier, to manually fix package.json before doing "npm install". As far as I am aware, the old 1.0.5 version can't be fixed in place on either GitHub or npm. If I publish a new 1.x that still won't prevent someone from doing a "npm install apg-exp@1.0.5" and getting the broken version. Sorry for not being a GitHub expert, but can you provide a little more detail on how I might publish a new 1.x that will hide 1.0.5 and not hide the new 2.0.0? Thanks.

Also, even though there are big changes in apg and apg-lib 3.0.0, there are virtually no feature or usage changes to apg-exp. Upgrading to apg-exp 2.0.0 shouldn't break any of the applications that use it.

**Comment by LinusU** [comment]
You cannot change a version that is already published, but you can publish a new 1.x version which most users will probably get since they most likely have `apg-exp` installed with `^1.0.5`. You can also deprecated the other versions with a message telling people to upgrade to the latest 1.x or 2.x, see: https://docs.npmjs.com/cli/deprecate

**Comment by ldthomas** [comment]
Thanks for the tip on deprecation. I've deprecated all versions < 2.0.0 with a message to upgrade or manually change the dependencies. Thanks for your help, and I hope you will be able to continue using apg-exp. I'm not sure about publishing a new 1.x version. You are probably right, but the deprecation message will have to do for just now.

**Comment by LinusU** [comment]
> I'm not sure about publishing a new 1.x version. You are probably right, but the deprecation message will have to do for just now.

Why not release a `1.1.4` with the only changes from `1.1.3` would be updated ranges in the package json?

**Comment by ldthomas** [comment]
A couple of naive questions; Just to be clear, if I create a version 1.1.4 and then "npm publish" will it fall in behind 2.0.0? Will a simple "npm install apg-exp" still get version 2.0.0? If it is that simple, I'll see if I can get that done in the next day or two.

**Comment by LinusU** [comment]
> if I create a version 1.1.4 and then "npm publish" will it fall in behind 2.0.0? Will a simple "npm install apg-exp" still get version 2.0.0?

That is correct 👍

**Comment by ldthomas** [comment]
OK, I have published apg-exp@1.1.4 which correctly installs apg@2.2.2 and apg-lib@2.1.5. All versions <=1.1.3 have a deprecation message.

However, "npm install apg-exp" installs the "latest" published version, not the highest version number. I had to publish apg-exp@2.0.1 as a follow up in order for the default install to install the new 2.0.1 version.

Anyway, thanks for pointing out the problem and your help. Let me know if you have any more problems with apg-exp.
